File: sesion-5/secure-snippets-js-csharp/js/5-path-traversal.js
/**
 * ✅ SEGURO: Validación de rutas y restricción a un directorio base
 */
const express = require('express');
const fs = require('fs');
const path = require('path');
const app = express();

const BASE_DIR = path.join(__dirname, 'files'); // restringido a carpeta files

app.get('/read', (req, res) => {
  const filename = path.basename(req.query.file || "");
  const filePath = path.join(BASE_DIR, filename);
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    res.type('text/plain').send(content);
  } catch (e) {
    res.status(404).send('No encontrado');
  }
});

app.listen(3002, () => console.log('Servidor seguro en :3002'));

File: sesion-5/secure-snippets-js-csharp/csharp/5-PathTraversal.cs
// ✅ SEGURO: Validación de rutas
using System;
using System.IO;

class PathSafe {
    static void Main() {
        Console.Write("Archivo a leer: ");
        var file = Path.GetFileName(Console.ReadLine()); // sanitizar input
        var baseDir = Path.Combine(Directory.GetCurrentDirectory(), "files");
        var path = Path.Combine(baseDir, file);
        if (File.Exists(path)) {
            Console.WriteLine(File.ReadAllText(path));
        } else {
            Console.WriteLine("No encontrado");
        }
    }
}
